### PR TITLE
feat: using the cookiecutter template

### PR DESCRIPTION
**Pull request by kryptonite303**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,38 +1,45 @@
-Role Name
-=========
+<<<<<<< HEAD
```
Comment (jasonwalsh) [inline README.md]: You have a merge conflict in your README.

File: README.md [hunk 2]
```diff
@@ -1,38 +1,45 @@
...
-         - { role: username.rolename, x: 42 }
+```yaml
+- hosts: all
+  roles:
+    - role: ansible-role-golang
+      vars:
+        versions:
+          - "1.7"
+          - "1.8"
+        url: http://boxes.10gen.com/build/go{{ item }}.linux-amd64.tar.gz
```
Comment (jasonwalsh) [inline README.md]: Where it `item` sourced from? I thought it was only available for use with `with_items`.

File: defaults/main.yml [hunk 3]
```diff
@@ -1,2 +1,5 @@
 ---
-# defaults file for ansible-role-golang
\ No newline at end of file
+url: http://boxes.10gen.com/build/go{{ item }}.linux-amd64.tar.gz
```
Comment (jasonwalsh) [inline defaults/main.yml]: Always prefix your variables with the role name (e.g., `golang_download_url`, `golang_versions`)

File: README.md (hunk 2 above)
Comment (kryptonite303) [reply README.md]: It's located in tasks:
`- name: Download Go
  unarchive:
    remote_src: true
    src: "{{ golang_download_url }}"
    dest: /opt/go{{ item }}
  loop: "{{ golang_versions }}"`
It loops through the `golang_versions` and inserts them into the url

Comment (jasonwalsh) [reply README.md]: That is confusing. I think the better approach would be to move the `golang_download_url` to the `vars/main.yml` file. The `vars` directory contains variables that *should* not be modified by the user.

**Pull request merged**